modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisFilterToSQL.java

```diff
@@ -92,6 +94,21 @@ protected void visitLiteralGeometry(Literal expression) throws IOException {
     }
 
     @Override
+    protected Object evaluateLiteral(Literal expression, Class target) {
+        Date converted = null;
+        if (target == PostGISDialect.XDate.class) {
+            converted = Converters.convert(expression.evaluate(null), Date.class);
```

Comment:
Thanks for providing that perspective. I didn't understand what was going on before, but now it makes sense - in a disconcerting, hackish way.